### ISSUE TITLE
As for the training question

### ISSUE THREAD

**Issue by YZGod666** [erdongsanshi/Fuss-Free-structure#4]
Hello, when I was training the data set of Shanghai Science and Technology A, I changed the num worker to 1 and the batch size to 2, and MAE and MSE became higher and higher when the model reached the 519th epoch in the training process. What happened?

1091 Comaprison 215.92798600878035 234.4008415177628
1092 2024-09-26 00:33:21,577 - INFO - save best mse 339.09 mae 215.93 model epoch 519
1093 Saving best model at 519 epoch
1094 2024-09-26 00:33:23,865 - INFO - -----Epoch 520/2000-----
1095 2024-09-26 00:33:40,779 - INFO - Epoch 520 Train, Loss: 49.94, OT Loss: -1.10e-07, Wass Distance: 189.49, OT obj value: 18.05, Count Loss: 48.70, TV Loss: 1.24, MSE: 93.85 MAE: 48.70, Cost 16.9 sec
1096 2024-09-26 00:33:54,434 - INFO - Epoch 520 Val, MSE: 391.01 MAE: 256.15, Cost 11.8 sec
1097 Comaprison 256.153854742155 215.92798600878035
1098 2024-09-26 00:33:54,437 - INFO - -----Epoch 521/2000-----
1099 2024-09-26 00:34:11,177 - INFO - Epoch 521 Train, Loss: 54.13, OT Loss: 5.68e-08, Wass Distance: 185.35, OT obj value: 19.23, Count Loss: 52.88, TV Loss: 1.25, MSE: 96.47 MAE: 52.88, Cost 16.7 sec
1100 2024-09-26 00:34:25,000 - INFO - Epoch 521 Val, MSE: 372.68 MAE: 237.01, Cost 12.1 sec
1101 Comaprison 237.00634506312045 215.92798600878035
1102 2024-09-26 00:34:25,003 - INFO - -----Epoch 522/2000-----
1103 2024-09-26 00:34:42,003 - INFO - Epoch 522 Train, Loss: 53.56, OT Loss: 4.18e-08, Wass Distance: 189.50, OT obj value: 19.05, Count Loss: 52.29, TV Loss: 1.27, MSE: 91.61 MAE: 52.29, Cost 17.0 sec
1104 2024-09-26 00:34:55,949 - INFO - Epoch 522 Val, MSE: 429.94 MAE: 293.94, Cost 12.2 sec
1105 Comaprison 293.9446882333729 215.92798600878035
1106 2024-09-26 00:34:55,952 - INFO - -----Epoch 523/2000-----
1107 2024-09-26 00:35:13,080 - INFO - Epoch 523 Train, Loss: 49.15, OT Loss: -1.06e-07, Wass Distance: 188.38, OT obj value: 20.13, Count Loss: 47.90, TV Loss: 1.25, MSE: 81.74 MAE: 47.90, Cost 17.1 sec
1108 2024-09-26 00:35:27,085 - INFO - Epoch 523 Val, MSE: 445.13 MAE: 296.38, Cost 12.1 sec
1109 Comaprison 296.3756193433489 215.92798600878035
1110 2024-09-26 00:35:27,088 - INFO - -----Epoch 524/2000-----
1111 2024-09-26 00:35:43,982 - INFO - Epoch 524 Train, Loss: 54.60, OT Loss: 8.40e-08, Wass Distance: 189.96, OT obj value: 17.99, Count Loss: 53.22, TV Loss: 1.38, MSE: 101.70 MAE: 53.22, Cost 16.9 sec
1112 2024-09-26 00:35:57,674 - INFO - Epoch 524 Val, MSE: 422.85 MAE: 285.41, Cost 11.9 sec
1113 Comaprison 285.40859899416074 215.92798600878035
1114 2024-09-26 00:35:57,677 - INFO - -----Epoch 525/2000-----
1115 2024-09-26 00:36:14,500 - INFO - Epoch 525 Train, Loss: 51.38, OT Loss: -1.40e-07, Wass Distance: 188.16, OT obj value: 19.18, Count Loss: 50.09, TV Loss: 1.29, MSE: 91.45 MAE: 50.09, Cost 16.8 sec
1116 2024-09-26 00:36:28,314 - INFO - Epoch 525 Val, MSE: 418.98 MAE: 285.00, Cost 11.8 sec
1117 Comaprison 284.99791058865225 215.92798600878035
1118 2024-09-26 00:36:28,317 - INFO - -----Epoch 526/2000-----
1119 2024-09-26 00:36:45,186 - INFO - Epoch 526 Train, Loss: 52.12, OT Loss: 7.79e-09, Wass Distance: 189.87, OT obj value: 17.89, Count Loss: 50.80, TV Loss: 1.31, MSE: 84.86 MAE: 50.80, Cost 16.9 sec
1120 2024-09-26 00:36:59,098 - INFO - Epoch 526 Val, MSE: 433.90 MAE: 293.67, Cost 12.2 sec
1121 Comaprison 293.6746255329677 215.92798600878035
1122 2024-09-26 00:36:59,101 - INFO - -----Epoch 527/2000-----
1123 2024-09-26 00:37:16,020 - INFO - Epoch 527 Train, Loss: 48.39, OT Loss: 1.18e-07, Wass Distance: 189.46, OT obj value: 19.40, Count Loss: 47.16, TV Loss: 1.23, MSE: 80.84 MAE: 47.16, Cost 16.9 sec
1124 2024-09-26 00:37:29,579 - INFO - Epoch 527 Val, MSE: 393.66 MAE: 265.10, Cost 11.7 sec
1125 Comaprison 265.1035563801671 215.92798600878035
1126 2024-09-26 00:37:29,582 - INFO - -----Epoch 528/2000-----
1127 2024-09-26 00:37:46,291 - INFO - Epoch 528 Train, Loss: 51.25, OT Loss: 6.23e-08, Wass Distance: 191.73, OT obj value: 19.58, Count Loss: 49.98, TV Loss: 1.27, MSE: 88.18 MAE: 49.98, Cost 16.7 sec
1128 2024-09-26 00:38:00,138 - INFO - Epoch 528 Val, MSE: 423.80 MAE: 295.30, Cost 12.0 sec
1129 Comaprison 295.30173149475684 215.92798600878035

**Comment by erdongsanshi** [comment]
It's normal to have ups and downs during training, but the whole training is a constant process of tuning, you can try training for a while and see the results. Just by adjusting batchsize and num worker I can't draw any conclusions.

**Comment by YZGod666** [comment]
However, I failed to achieve the counting effect mentioned in your paper by using a single GPU on my 3060 graphics card. Various indicators of the model fluctuated but did not decline during the training until the end of the whole training process

**Comment by erdongsanshi** [comment]
However, I failed to achieve the counting effect mentioned in your paper by using a single GPU on my 3060 graphics card. Various indicators of the model fluctuated but did not decline during the training until the end of the whole training process

**Comment by erdongsanshi** [comment]
Can we get your training logs? Have you tried to change hyperparameters, or check whether backbone's pre-training model loads properly, others in the community have not encountered this situation, need more details

**Comment by YZGod666** [comment]
[output.log](https://github.com/user-attachments/files/17171985/output.log)
hi! Thank you for taking the time to answer my questions. This is my training log, which has been uploaded through the attachment

**Comment by erdongsanshi** [comment]
Sorry, I have been busy recently, so my reply is slow. There is no hyperparameter setting information in your log information. Can you provide further hyperparameter information? And have you tried adjusting hyperparameters, like batch or lr, etc? You can also check to see if the pre-trained model is loaded correctly

**Comment by YZGod666** [comment]
![image](https://github.com/user-attachments/assets/de5f4860-6e97-417a-beb1-78ca2745fa9c)
This is the part about hyperparameter setting at the beginning of the training log. Thank you very much for your help

**Comment by erdongsanshi** [comment]
According to what you provided, I trained the model and found that mae was indeed high when batch was 2, and my analysis was when bs was 2. Because the gradient was calculated based on only two samples, the gradient estimation of each batch would be very unstable. This may lead to the model weight updating in an inconsistent direction, or even in the wrong direction, making it difficult for the model to converge to a good solution. I think you can try two ways, one: appropriately reduce the learning rate. Two: Increase bs. The first method I did not experiment with, just based on experience, the second method, when I set bs to 4, mae has already converged to a respectable value. I hope I can help you.

**Comment by YZGod666** [comment]
Ok, thanks for your reply, I will try training again with bitch size set to 4
